src/compiler/emitter.ts

```diff
@@ -1257,26 +1257,34 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, ge
 
                     // Children
                     if (children) {
-                        for (let i = 0; i < children.length; i++) {
-                            // Don't emit empty expressions
-                            if (children[i].kind === SyntaxKind.JsxExpression && !((<JsxExpression>children[i]).expression)) {
-                                continue;
+                        const isChildEmittable = (child: JsxChild): boolean => {

```

Comment:
Why do you think we don't need the function? It was suggested by a previous
reviewer.  I'll move the function outside as a helper.

Please let me know if you have any other concerns.

On Monday, January 25, 2016, Mohamed Hegazy <notifications@github.com>
wrote:

> In src/compiler/emitter.ts
> <https://github.com/Microsoft/TypeScript/pull/6587#discussion_r50770713>:
>
> > @@ -1257,26 +1257,34 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, ge
> >
> >                      // Children
> >                      if (children) {
> > -                        for (let i = 0; i < children.length; i++) {
> > -                            // Don't emit empty expressions
> > -                            if (children[i].kind === SyntaxKind.JsxExpression && !((<JsxExpression>children[i]).expression)) {
> > -                                continue;
> > +                        const isChildEmittable = (child: JsxChild): boolean => {
>
> i do nto think we need this function, but even if we did we should not
> create one every time we enter this block.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Microsoft/TypeScript/pull/6587/files#r50770713>.
>